experiments/AtmosLES/baroclinicwave.jl

```diff
@@ -0,0 +1,332 @@
+  Lp    = FT(6e5)              ## Perturbation parameter (radius)
+  Lp2   = Lp*Lp              
+  xc    = FT(2e6)              ## Streamwise center of perturbation
+  yc    = FT(2.5e6)             ## Spanwise center of perturbation
+  gamma_lapse = FT(5/1000)    ## Γ Lapse Rate
+  Ω     = FT(7.292e-5)         ## Rotation rate [rad/s]
+  f0    = 2Ω/sqrt(2)           ## 
+  beta0 = f0/_planet_radius    ##  
+  beta0 = -zero(FT)
+  b     = FT(2)
+  b2    = b*b
+  u0    = FT(35)
+  Ly    = FT(6e6)
+  T0    = FT(288)
+  T_ref = T0                   
+  x0    = FT(2e7)
+  p00   = FT(1e5)              ## Surface pressure
+
+  ## Step 1: Get current coordinate value by unpacking nodal coordinates from aux state
+  eta = FT(1e-7)
+  temp = FT(300)
+  ## Step 2: Define functions for initial condition temperature and geopotential distributions
+  ## These are written in terms of the pressure coordinate η = p/pₛ
+
+  ### Unpack initial conditions (solved by iterating for η)
+  tolerance = FT(1e-10)
+  eta, temp = baroclinic_instability_cube!(eta, 
+                                           temp, 
+                                           tolerance, 
+                                           (x,y,z),
+                                           f0, 
+                                           beta0, 
+                                           u0, 
+                                           T0, 
+                                           gamma_lapse, 
+                                           gravity, 
+                                           R_gas, 
+                                           cp)
+  eta = min(eta,FT(1))
+  eta = max(eta,FT(0))
+  ### η = p/p_s
+  logeta = log(eta)
+  T=FT(temp)
+  press = p00*eta
+  theta = T *(p00/press)^(R_gas/cp)
+  rho = press/(R_gas*T)
+  thetaref = T_ref * (1 - gamma_lapse*z/T0)^(-gravity/(cp*gamma_lapse))
+  rhoref = p00/(T0*R_gas) * (1 - gamma_lapse*z/T0)^(gravity/(R_gas*gamma_lapse) - 1)
+
+  ### Balanced Flow
+  u = -u0*(sinpi(y/Ly))^2  * logeta * exp(-logeta*logeta/b2)
+
+  ### Perturbation of the balanced flow
+  rc2 = (x-xc)^2 + (y-yc)^2
+  du = up*exp(-rc2/Lp2)
+    
+  ### Primitive variables
+  u⃗ = SVector{3,FT}(u+du,0,0)
+  e_kin = FT(1/2)*sum(abs2.(u⃗))
+  e_pot = gravity * z
+
+  ### Assign state variables for initial condition
+  state.ρ = rho
+  state.ρu = rho .* u⃗
+  state.ρe = rho * total_energy(param_set, e_kin, e_pot, T)
+end
+
+function config_baroclinicwave(FT, N, resolution, xmax, ymax, zmax)
+
+    ics = init_baroclinicwave!     # Initial conditions
+    C_smag = FT(0.30)     # Smagorinsky coefficient
+    
+    # Assemble source components
+    source = (
+        Gravity(),
+    )
+
+    # Choose default IMEX solver
+    ode_solver_type = ClimateMachine.IMEXSolverType()
+
+    # Assemble model components
+    model = AtmosModel{FT}(
+        AtmosLESConfigType,
+        param_set;
+        turbulence = SmagorinskyLilly{FT}(C_smag),
```

Comment:
Although it's ok to use Smagorinsky for the sake of testing  LES, eventually this should be replaced by a 4th or 6th order (maybe higher?) hyper-diffusion for numerical stability purposes.